[12:02:11,927] [main] DEBUG (MutableMetricsFactory.java:42) - field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.loginSuccess with annotation @org.apache.hadoop.metrics2.annotation.Metric(valueName=Time, value=[Rate of successful kerberos logins and latency (milliseconds)], about=, type=DEFAULT, always=false, sampleName=Ops)
[12:02:11,945] [main] DEBUG (MutableMetricsFactory.java:42) - field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.loginFailure with annotation @org.apache.hadoop.metrics2.annotation.Metric(valueName=Time, value=[Rate of failed kerberos logins and latency (milliseconds)], about=, type=DEFAULT, always=false, sampleName=Ops)
[12:02:11,947] [main] DEBUG (MetricsSystemImpl.java:220) - UgiMetrics, User and group related metrics
[12:02:12,215] [main] DEBUG (Groups.java:180) -  Creating new Groups object
[12:02:12,217] [main] DEBUG (NativeCodeLoader.java:46) - Trying to load the custom-built native-hadoop library...
[12:02:12,218] [main] DEBUG (NativeCodeLoader.java:55) - Failed to load native-hadoop with error: java.lang.UnsatisfiedLinkError: no hadoop in java.library.path
[12:02:12,218] [main] DEBUG (NativeCodeLoader.java:56) - java.library.path=/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
[12:02:12,218] [main]  WARN (NativeCodeLoader.java:62) - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[12:02:12,218] [main] DEBUG (JniBasedUnixGroupsMappingWithFallback.java:40) - Falling back to shell based
[12:02:12,219] [main] DEBUG (JniBasedUnixGroupsMappingWithFallback.java:44) - Group mapping impl=org.apache.hadoop.security.ShellBasedUnixGroupsMapping
[12:02:12,219] [main] DEBUG (Groups.java:66) - Group mapping impl=org.apache.hadoop.security.JniBasedUnixGroupsMappingWithFallback; cacheTimeout=300000
[12:02:12,223] [main] DEBUG (UserGroupInformation.java:177) - hadoop login
[12:02:12,223] [main] DEBUG (UserGroupInformation.java:126) - hadoop login commit
[12:02:12,227] [main] DEBUG (UserGroupInformation.java:156) - using local user:UnixPrincipal: kylin
[12:02:12,228] [main] DEBUG (UserGroupInformation.java:703) - UGI loginUser:kylin (auth:SIMPLE)
[12:02:12,337] [main] DEBUG (Shell.java:243) - Failed to detect a valid hadoop home directory
java.io.IOException: HADOOP_HOME or hadoop.home.dir are not set.
	at org.apache.hadoop.util.Shell.checkHadoopHome(Shell.java:225)
	at org.apache.hadoop.util.Shell.<clinit>(Shell.java:250)
	at org.apache.hadoop.util.StringUtils.<clinit>(StringUtils.java:76)
	at org.apache.hadoop.conf.Configuration.getStrings(Configuration.java:1514)
	at org.apache.hadoop.hbase.zookeeper.ZKConfig.makeZKProps(ZKConfig.java:113)
	at org.apache.hadoop.hbase.zookeeper.ZKConfig.getZKQuorumServersString(ZKConfig.java:259)
	at org.apache.hadoop.hbase.zookeeper.ZooKeeperWatcher.<init>(ZooKeeperWatcher.java:159)
	at org.apache.hadoop.hbase.zookeeper.ZooKeeperWatcher.<init>(ZooKeeperWatcher.java:134)
	at org.apache.hadoop.hbase.client.ZooKeeperKeepAliveConnection.<init>(ZooKeeperKeepAliveConnection.java:43)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getKeepAliveZooKeeperWatcher(HConnectionManager.java:1748)
	at org.apache.hadoop.hbase.client.ZooKeeperRegistry.getClusterId(ZooKeeperRegistry.java:82)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.retrieveClusterId(HConnectionManager.java:839)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.<init>(HConnectionManager.java:642)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at org.apache.hadoop.hbase.client.HConnectionManager.createConnection(HConnectionManager.java:411)
	at org.apache.hadoop.hbase.client.HConnectionManager.createConnection(HConnectionManager.java:390)
	at org.apache.hadoop.hbase.client.HConnectionManager.getConnection(HConnectionManager.java:271)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:32)
[12:02:12,354] [main] DEBUG (Shell.java:326) - setsid exited with exit code 0
[12:02:12,368] [main]  INFO (RecoverableZooKeeper.java:121) - Process identifier=hconnection-0x72f5ba2c connecting to ZooKeeper ensemble=localhost:2181
[12:02:12,375] [main]  INFO (Environment.java:100) - Client environment:zookeeper.version=3.4.6-1569965, built on 02/20/2014 09:09 GMT
[12:02:12,376] [main]  INFO (Environment.java:100) - Client environment:host.name=localhost
[12:02:12,376] [main]  INFO (Environment.java:100) - Client environment:java.version=1.7.0_60
[12:02:12,376] [main]  INFO (Environment.java:100) - Client environment:java.vendor=Oracle Corporation
[12:02:12,376] [main]  INFO (Environment.java:100) - Client environment:java.home=/usr/java/jdk1.7.0_60/jre
[12:02:12,377] [main]  INFO (Environment.java:100) - Client environment:java.class.path=/home/kylin/project/data/hbase-quickstart/target/classes:/home/kylin/.m2/repository/org/apache/hbase/hbase-client/0.98.8-hadoop2/hbase-client-0.98.8-hadoop2.jar:/home/kylin/.m2/repository/org/apache/hbase/hbase-annotations/0.98.8-hadoop2/hbase-annotations-0.98.8-hadoop2.jar:/usr/java/jdk1.7.0_60/lib/tools.jar:/home/kylin/.m2/repository/log4j/log4j/1.2.17/log4j-1.2.17.jar:/home/kylin/.m2/repository/org/apache/hbase/hbase-common/0.98.8-hadoop2/hbase-common-0.98.8-hadoop2.jar:/home/kylin/.m2/repository/commons-collections/commons-collections/3.2.1/commons-collections-3.2.1.jar:/home/kylin/.m2/repository/org/apache/hbase/hbase-protocol/0.98.8-hadoop2/hbase-protocol-0.98.8-hadoop2.jar:/home/kylin/.m2/repository/commons-codec/commons-codec/1.7/commons-codec-1.7.jar:/home/kylin/.m2/repository/commons-io/commons-io/2.4/commons-io-2.4.jar:/home/kylin/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar:/home/kylin/.m2/repository/commons-logging/commons-logging/1.1.1/commons-logging-1.1.1.jar:/home/kylin/.m2/repository/com/google/guava/guava/12.0.1/guava-12.0.1.jar:/home/kylin/.m2/repository/com/google/code/findbugs/jsr305/1.3.9/jsr305-1.3.9.jar:/home/kylin/.m2/repository/com/google/protobuf/protobuf-java/2.5.0/protobuf-java-2.5.0.jar:/home/kylin/.m2/repository/io/netty/netty/3.6.6.Final/netty-3.6.6.Final.jar:/home/kylin/.m2/repository/org/apache/zookeeper/zookeeper/3.4.6/zookeeper-3.4.6.jar:/home/kylin/.m2/repository/org/slf4j/slf4j-api/1.6.1/slf4j-api-1.6.1.jar:/home/kylin/.m2/repository/org/slf4j/slf4j-log4j12/1.6.1/slf4j-log4j12-1.6.1.jar:/home/kylin/.m2/repository/org/cloudera/htrace/htrace-core/2.04/htrace-core-2.04.jar:/home/kylin/.m2/repository/org/mortbay/jetty/jetty-util/6.1.26/jetty-util-6.1.26.jar:/home/kylin/.m2/repository/org/codehaus/jackson/jackson-mapper-asl/1.8.8/jackson-mapper-asl-1.8.8.jar:/home/kylin/.m2/repository/org/codehaus/jackson/jackson-core-asl/1.8.8/jackson-core-asl-1.8.8.jar:/home/kylin/.m2/repository/org/jruby/joni/joni/2.1.2/joni-2.1.2.jar:/home/kylin/.m2/repository/org/jruby/jcodings/jcodings/1.0.8/jcodings-1.0.8.jar:/home/kylin/.m2/repository/org/apache/hadoop/hadoop-common/2.2.0/hadoop-common-2.2.0.jar:/home/kylin/.m2/repository/org/apache/hadoop/hadoop-annotations/2.2.0/hadoop-annotations-2.2.0.jar:/home/kylin/.m2/repository/commons-cli/commons-cli/1.2/commons-cli-1.2.jar:/home/kylin/.m2/repository/org/apache/commons/commons-math/2.1/commons-math-2.1.jar:/home/kylin/.m2/repository/xmlenc/xmlenc/0.52/xmlenc-0.52.jar:/home/kylin/.m2/repository/commons-httpclient/commons-httpclient/3.1/commons-httpclient-3.1.jar:/home/kylin/.m2/repository/commons-net/commons-net/3.1/commons-net-3.1.jar:/home/kylin/.m2/repository/org/mortbay/jetty/jetty/6.1.26/jetty-6.1.26.jar:/home/kylin/.m2/repository/com/sun/jersey/jersey-core/1.9/jersey-core-1.9.jar:/home/kylin/.m2/repository/com/sun/jersey/jersey-json/1.9/jersey-json-1.9.jar:/home/kylin/.m2/repository/org/codehaus/jettison/jettison/1.1/jettison-1.1.jar:/home/kylin/.m2/repository/stax/stax-api/1.0.1/stax-api-1.0.1.jar:/home/kylin/.m2/repository/com/sun/xml/bind/jaxb-impl/2.2.3-1/jaxb-impl-2.2.3-1.jar:/home/kylin/.m2/repository/javax/xml/bind/jaxb-api/2.2.2/jaxb-api-2.2.2.jar:/home/kylin/.m2/repository/javax/activation/activation/1.1/activation-1.1.jar:/home/kylin/.m2/repository/org/codehaus/jackson/jackson-jaxrs/1.8.3/jackson-jaxrs-1.8.3.jar:/home/kylin/.m2/repository/org/codehaus/jackson/jackson-xc/1.8.3/jackson-xc-1.8.3.jar:/home/kylin/.m2/repository/commons-el/commons-el/1.0/commons-el-1.0.jar:/home/kylin/.m2/repository/net/java/dev/jets3t/jets3t/0.6.1/jets3t-0.6.1.jar:/home/kylin/.m2/repository/commons-configuration/commons-configuration/1.6/commons-configuration-1.6.jar:/home/kylin/.m2/repository/commons-digester/commons-digester/1.8/commons-digester-1.8.jar:/home/kylin/.m2/repository/commons-beanutils/commons-beanutils/1.7.0/commons-beanutils-1.7.0.jar:/home/kylin/.m2/repository/commons-beanutils/commons-beanutils-core/1.8.0/commons-beanutils-core-1.8.0.jar:/home/kylin/.m2/repository/org/apache/avro/avro/1.7.4/avro-1.7.4.jar:/home/kylin/.m2/repository/com/thoughtworks/paranamer/paranamer/2.3/paranamer-2.3.jar:/home/kylin/.m2/repository/org/xerial/snappy/snappy-java/1.0.4.1/snappy-java-1.0.4.1.jar:/home/kylin/.m2/repository/com/jcraft/jsch/0.1.42/jsch-0.1.42.jar:/home/kylin/.m2/repository/org/apache/commons/commons-compress/1.4.1/commons-compress-1.4.1.jar:/home/kylin/.m2/repository/org/tukaani/xz/1.0/xz-1.0.jar:/home/kylin/.m2/repository/org/apache/hadoop/hadoop-auth/2.2.0/hadoop-auth-2.2.0.jar:/home/kylin/.m2/repository/org/apache/hadoop/hadoop-mapreduce-client-core/2.2.0/hadoop-mapreduce-client-core-2.2.0.jar:/home/kylin/.m2/repository/org/apache/hadoop/hadoop-yarn-common/2.2.0/hadoop-yarn-common-2.2.0.jar:/home/kylin/.m2/repository/org/apache/hadoop/hadoop-yarn-api/2.2.0/hadoop-yarn-api-2.2.0.jar:/home/kylin/.m2/repository/com/google/inject/guice/3.0/guice-3.0.jar:/home/kylin/.m2/repository/javax/inject/javax.inject/1/javax.inject-1.jar:/home/kylin/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar:/home/kylin/.m2/repository/com/sun/jersey/jersey-server/1.9/jersey-server-1.9.jar:/home/kylin/.m2/repository/asm/asm/3.1/asm-3.1.jar:/home/kylin/.m2/repository/com/sun/jersey/contribs/jersey-guice/1.9/jersey-guice-1.9.jar:/home/kylin/.m2/repository/com/google/inject/extensions/guice-servlet/3.0/guice-servlet-3.0.jar:/home/kylin/.m2/repository/com/github/stephenc/findbugs/findbugs-annotations/1.3.9-1/findbugs-annotations-1.3.9-1.jar
[12:02:12,377] [main]  INFO (Environment.java:100) - Client environment:java.library.path=/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
[12:02:12,377] [main]  INFO (Environment.java:100) - Client environment:java.io.tmpdir=/tmp
[12:02:12,377] [main]  INFO (Environment.java:100) - Client environment:java.compiler=<NA>
[12:02:12,378] [main]  INFO (Environment.java:100) - Client environment:os.name=Linux
[12:02:12,378] [main]  INFO (Environment.java:100) - Client environment:os.arch=amd64
[12:02:12,378] [main]  INFO (Environment.java:100) - Client environment:os.version=2.6.32-431.20.3.el6.x86_64
[12:02:12,378] [main]  INFO (Environment.java:100) - Client environment:user.name=kylin
[12:02:12,378] [main]  INFO (Environment.java:100) - Client environment:user.home=/home/kylin
[12:02:12,379] [main]  INFO (Environment.java:100) - Client environment:user.dir=/home/kylin/project/data/hbase-quickstart
[12:02:12,380] [main]  INFO (ZooKeeper.java:438) - Initiating client connection, connectString=localhost:2181 sessionTimeout=90000 watcher=hconnection-0x72f5ba2c, quorum=localhost:2181, baseZNode=/hbase
[12:02:12,387] [main] DEBUG (ClientCnxn.java:102) - zookeeper.disableAutoWatchReset is false
[12:02:12,404] [main-SendThread(localhost:2181)]  INFO (ClientCnxn.java:975) - Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error)
[12:02:12,410] [main-SendThread(localhost:2181)]  INFO (ClientCnxn.java:852) - Socket connection established to localhost/127.0.0.1:2181, initiating session
[12:02:12,411] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:892) - Session establishment request sent on localhost/127.0.0.1:2181
[12:02:12,475] [main-SendThread(localhost:2181)]  INFO (ClientCnxn.java:1235) - Session establishment complete on server localhost/127.0.0.1:2181, sessionid = 0x14a08a8e944000a, negotiated timeout = 40000
[12:02:12,476] [main-EventThread] DEBUG (ZooKeeperWatcher.java:312) - hconnection-0x72f5ba2c, quorum=localhost:2181, baseZNode=/hbase Received ZooKeeper Event, type=None, state=SyncConnected, path=null
[12:02:12,480] [main-EventThread] DEBUG (ZooKeeperWatcher.java:389) - hconnection-0x72f5ba2c-0x14a08a8e944000a connected
[12:02:12,482] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 1,3  replyHeader:: 1,463,0  request:: '/hbase/hbaseid,F  response:: s{13,322,1417408327474,1417484507019,1,0,0,0,67,0,13} 
[12:02:12,488] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 2,4  replyHeader:: 2,463,0  request:: '/hbase/hbaseid,F  response:: #ffffffff000146d61737465723a3533343232ffffffd56cfffffff7ffffff90ffffffb5ffffffaeffffffe7ffffffd550425546a2463613462646636372d623533312d346632382d626134622d316436643333636632636263,s{13,322,1417408327474,1417484507019,1,0,0,0,67,0,13} 
[12:02:12,678] [main] DEBUG (RpcClient.java:1283) - Codec=org.apache.hadoop.hbase.codec.KeyValueCodec@24c96a1a, compressor=null, tcpKeepAlive=true, tcpNoDelay=true, maxIdleTime=10000, maxRetries=0, fallbackAllowed=false, ping interval=60000ms, bind address=null
[12:02:12,708] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 3,4  replyHeader:: 3,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:02:12,857] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:02:12,862] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:38776
[12:02:12,865] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:02:12,866] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed
[12:02:13,174] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 4,4  replyHeader:: 4,463,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[12:02:13,189] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 5,4  replyHeader:: 5,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:02:13,190] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:02:13,190] [main] DEBUG (RpcClient.java:850) - Not trying to connect to localhost/127.0.0.1:38776 this server is in the failed servers list
[12:02:13,191] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: This server is in the failed servers list: localhost/127.0.0.1:38776
org.apache.hadoop.hbase.ipc.RpcClient$FailedServerException: This server is in the failed servers list: localhost/127.0.0.1:38776
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:853)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:02:13,193] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed
[12:02:13,703] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 6,4  replyHeader:: 6,463,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[12:02:13,707] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 7,4  replyHeader:: 7,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:02:13,708] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:02:13,709] [main] DEBUG (RpcClient.java:850) - Not trying to connect to localhost/127.0.0.1:38776 this server is in the failed servers list
[12:02:13,709] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: This server is in the failed servers list: localhost/127.0.0.1:38776
org.apache.hadoop.hbase.ipc.RpcClient$FailedServerException: This server is in the failed servers list: localhost/127.0.0.1:38776
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:853)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:02:13,711] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed
[12:02:14,720] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 8,4  replyHeader:: 8,463,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[12:02:14,724] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 9,4  replyHeader:: 9,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:02:14,724] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:02:14,725] [main] DEBUG (RpcClient.java:850) - Not trying to connect to localhost/127.0.0.1:38776 this server is in the failed servers list
[12:02:14,725] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: This server is in the failed servers list: localhost/127.0.0.1:38776
org.apache.hadoop.hbase.ipc.RpcClient$FailedServerException: This server is in the failed servers list: localhost/127.0.0.1:38776
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:853)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:02:14,725] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed
[12:02:16,734] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 10,4  replyHeader:: 10,463,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[12:02:16,737] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 11,4  replyHeader:: 11,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:02:16,738] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:02:16,738] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:38776
[12:02:16,739] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:02:16,740] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed
[12:02:20,763] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 12,4  replyHeader:: 12,463,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[12:02:20,768] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 13,4  replyHeader:: 13,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:02:20,769] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:02:20,770] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:38776
[12:02:20,771] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:02:20,773] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed
[12:02:30,858] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 14,4  replyHeader:: 14,463,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[12:02:30,859] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a08a8e944000a after 4ms
[12:02:30,863] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 15,4  replyHeader:: 15,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:02:30,864] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:02:30,865] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:38776
[12:02:30,866] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:02:30,867] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed
[12:02:40,935] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 16,4  replyHeader:: 16,463,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[12:02:40,937] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a08a8e944000a after 6ms
[12:02:40,942] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 17,4  replyHeader:: 17,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:02:40,943] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:02:40,944] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:38776
[12:02:40,950] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:02:40,950] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed
[12:02:50,981] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 18,4  replyHeader:: 18,463,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[12:02:50,983] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a08a8e944000a after 5ms
[12:02:50,986] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 19,4  replyHeader:: 19,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:02:50,987] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:02:50,988] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:38776
[12:02:50,989] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:02:50,991] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed
[12:03:01,035] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 20,4  replyHeader:: 20,463,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[12:03:01,036] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a08a8e944000a after 4ms
[12:03:01,039] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a08a8e944000a, packet:: clientPath:null serverPath:null finished:false header:: 21,4  replyHeader:: 21,463,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a3338373736ffffffb5ffffffed4ffffffe330ffffffdb54ffffffb450425546a16a96c6f63616c686f737410fffffff8ffffffae218ffffffddffffffe7ffffffa3ffffffc5ffffffa029100,s{337,337,1417484512355,1417484512355,0,0,0,0,61,0,337} 
[12:03:01,040] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[12:03:01,041] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:38776
[12:03:01,042] [main] DEBUG (RpcClient.java:1009) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closing ipc connection to localhost/127.0.0.1:38776: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.examples.ConnectionRefuseReproduce.main(ConnectionRefuseReproduce.java:37)
[12:03:01,043] [main] DEBUG (RpcClient.java:1017) - IPC Client (1455309687) connection to localhost/127.0.0.1:38776 from kylin: closed

